### DISCUSSION TITLE
Ready to work on download file

### DISCUSSION THREAD

**Discussion by Kaloyan96** [google/j2cl#194] [Q&A]
Is the there a file that can be downloaded from somewhere that can convert from java to JS?

And no, I don't need a set of instructions with a raw source code that then must be assembled into a working file, I need that working file. I've wasted over a week trying to "compile it myself".

I just need the file.
The one that is executable at the end.
Not the raw source code that I need to assemble myself.
Just the file.

**Comment by treblereel** [comment]
If you need to compile a Java application into JavaScript with J2CL, you also need to use the Closure Compiler. If you prefer Bazel, you can check out how to do it in j2cl/samples/helloworld. If you prefer Maven, look at https://github.com/Vertispan/j2clmavenplugin and a simple project example at https://github.com/treblereel/j2cl-tests.

I also recommend that you read the list of J2CL limitations at https://github.com/google/j2cl/blob/master/docs/limitations.md.
